### DISCUSSION TITLE
Terraform Provider for Linkerd CRDs

### DISCUSSION THREAD

**Discussion by ghost** [linkerd/linkerd2#8253] [Q&A]
Hi,

I know linkerd has two CRDs for traffic management: `Server` and `ServerAuthorization`. I can create custom `Server`s in Terraform via the kubernetes_manifest resource in the `kubernetes` provider. An abridged snippet would look something like this
```hcl2
resource "kubernetes_manifest" "linkerd_example_server" {
  manifest = {
    "apiVersion" = "policy.linkerd.io/v1beta1"
    "kind"       = "Server"
...
```

- Is there a terraform provider that exists that would let me create such `Server` and `ServerAuthorization` objects natively? You'd get a bunch of advantages like being able to natively refer to servers in serverauthorizations, do better validation of server objects, etc.
- Would there be interest in consuming such a terraform module if one was made (if nothing like this exists)?

Thanks

**Comment by olix0r** [comment]
Is it sufficient to use the helm provider as described here? https://github.com/linkerd/linkerd2/discussions/6230#discussioncomment-874629

In general, I'm not opposed to tighter terraform integration, but it really depends on the maintenance overhead. If the helm approach doesn't work, I'd recommend starting a separate repo that demonstrates it, and if there's broader community interest we can consider taking it on as an officially supported feature.

**Reply by ghost** [reply]
I think the helm chart installs linkerd itself, whereas I'd like to just provision Servers and ServerAuthorizations. But looking around, it looks like nothing like this exists. I'll attempt to build a provider because it's useful for us to try and provision this specific resource in terraform.
